js/plugins/trimTimeline.js

```diff
@@ -0,0 +1,183 @@
+        var second = ticksToShowKink[index + 1];
+
+        if (second - first > tolerance) {
+          breakTimelineForKink.push(index + 1)
+        } 
+      }
+    });
+  }
+
+  
+
+  var xScale = d3.time.scale()
+      .domain([margin.left,  width - margin.right])
+      .range([margin.left, width - margin.right]);
+
+  var shiftAtBreak = 0; //make space for adding the kink
+
+  for (var i = 0; i < ticksToShowKink.length; i++) {
+    if (breakTimelineForKink.indexOf(i) > -1) {
+      shiftAtBreak += 30;
+    }
+    //calculate the position of the tick to be shown after trimming
+    tickCoordiantesKink.push(margin.left + shiftAtBreak + (i * ((width - 3 * margin.right - margin.left) / (ticksToShowKink.length - 1 ))))
+  }
+
+  var xAxis = d3.svg.axis()
+    .scale(xScale)
+    .orient("top")
+    .tickValues(tickCoordiantesKink)
+    .tickFormat(function(d, i) { return formatTime(daysToTimeObject(ticksToShowKink[i]), zoomLevel) });
+
+  // Add the trimmmed axis
+  svg.append("g") 
+      .attr("class", "x axis")
+      .attr("transform", "translate(0,20)")
+      .call(xAxis);
+
+  var kink = svg.append("g")
+        .attr("class", "kink")
+
+  for (var i = 0; i < ticksToShowKink.length; i++) {
+    //add kinks at the places where timeline is cut
+    if (breakTimelineForKink.indexOf(i) > -1) {
+      shiftAtBreak += 30;
+      var kinkPosition= (shiftAtBreak + (i * ((width - margin.right - margin.left) / (ticksToShowKink.length)))); 
+
+      kink.append("rect")
+        .attr("height", 10)
+        .attr("width", 35)
+        .attr("class","kink-bg")
+        .attr("x", 70 + kinkPosition)
+        .attr("y", 15)
+        .attr("fill", "white");
+
+      kink.append("path")
+        .attr("d", kinkLine(kinkLineData))
+        .attr("stroke", "black")
+        .attr("stroke-width", 1)
+        .attr("fill", "none")
+        .attr("class","kink-line")
+        .attr("transform", "translate("+ kinkPosition +",20)");
+    }
+  }
+
+  function getLowerBoundIndex(ele) {
+    //returns the lower index of the indices between which the element, ele lies in an array
+    //based on binary search
+
+    var low = 0;
+    var high = ticksToShowKink.length - 1;
+
+    while(low < high){
+      var mid = Math.round((low + high)/2);
+      if (ticksToShowKink[mid] > ele) {
+        high = mid - 1;
+      } else {
+        low = mid;
+      }
+    }
+
+    return low;
+  }
+
+  function getXPosAdjustedForKink(x) {
+    //returns updated x positions for the data elements according to th trimmed timeline    
+    var first = getLowerBoundIndex(x.starting_time);
+    var second = first + 1;
+
+    if (second > ticksToShowKink.length - 1) {
+      return tickCoordiantesKink[first];
+    }
+
+    return tickCoordiantesKink[first] + (x.starting_time - ticksToShowKink[first]) * (tickCoordiantesKink[second] - tickCoordiantesKink[first]) / (ticksToShowKink[second] - ticksToShowKink[first]);
+  }
+
+  d3.selectAll(".timelineDataPoint").attr("cx", function(x) {
```

Comment:
It's not necessary to add the `timelineDataPoint` class. You could do something like `d3.selectAll("[id^=timelineItem]")`. It does make sense to have a `timelineItem` class, but since we are trying to use `d3-timeline` it's best to avoid making edits to that code unless absolutely necessary.